### ISSUE TITLE
OMEdit help menu not working properly on Linux

### ISSUE THREAD

**Issue by dietmarw** [OpenModelica/OpenModelica#8165]
On Linux (last tested on OpenModelica 1.19.0~dev-388-gb35aa9c) the help menu entries are only partially working. I doubt that is connected with the default viewer settings since some work. 

Here the list with check marks for thos that work on my machine:
 - [ ] OpenModelica Users Guide F1 (F1 also does nothing)
 - [ ] OpenModelica UsersGuide (PDF)  
 - [ ] OpenModelica System Documentation
 - [x] OpenModelica Scripting Documentation
 - [x] Modelica Documentation
 - [x] OMSimulator Users Guide (**Note the typo: it should be User's Guide**)
 - [ ] OpenMoelica TLM Simulator Documentation
 - [x] About OMEdit

**Comment by casella** [comment]
@adeas31, @sjoelund, I guess this should be fixed for the next release. Can you please check?

**Comment by adeas31** [comment]
See also #7857. I think the first 3 should be fixed if you install omc-doc package.

`OpenModelica TLM Simulator Documentation` perhaps need special handling. At least for the Windows installer I copy it from `OMSimulator/OMTLMSimulator/Documentation/TLMPlugin.pdf` and put it in `<INSTALL-DIR>/OMTLMSimulator/bin`

**Comment by dietmarw** [comment]
So installing omc-doc changes the status to:
  * [x]  OpenModelica Users Guide F1 (F1 also does nothing)
  * []  OpenModelica UsersGuide (PDF)
  * [x]  OpenModelica System Documentation
  * [x]  OpenModelica Scripting Documentation
  * [x]  Modelica Documentation
  * [x]  OMSimulator Users Guide (**Note the typo: it should be User's Guide**)
  * [ ]  OpenMoelica TLM Simulator Documentation
  * [x]  About OMEdit

So the second PDF item does not get opened while the third item (also PDF) does.
As @sjoelund also noted in #7857, the users guide shipped in omc-doc is completely outdated and should really be replaced. Also if OMEdit links to resources available in the optional package `omc-doc` that this package should also be added to the dependencies.

**Comment by dietmarw** [comment]
Also not the typos in the menu entries should get corrected to "User's..."

**Comment by casella** [comment]
@adeas31 should we do something to keep omc-doc updated (and mention it in the [installation guide](https://openmodelica.org/download/download-linux)) or should we just copy those documents in some proper place? 

I'm not sure having a separate debian package for documentation makes much sense, but maybe that's commonplace use? @dietmarw what would you suggest?

**Comment by dietmarw** [comment]
The question is where else is the content of omc-doc used/linked to. If only OMEdit then it makes sense to make (the updated version) part of OMEdit. Or simply link to the updated online-resources.

**Comment by adeas31** [comment]
> @adeas31 should we do something to keep omc-doc updated (and mention it in the installation guide) or should we just copy those documents in some proper place?

That's a question that Martin should answer. I basically suggested him the same (in #7857) what you and @dietmarw are saying here i.e., either install omc-doc as part of core package or at least put some instructions on the installation page.

**Comment by sjoelund** [comment]
I think if omc-doc is missing, we could just open the webpage

**Comment by sjoelund** [comment]
As for it being a separate package, that's common and in our case it's a separate tarball. I do want to change that though, since it's hard to keep it in sync if it's separate

**Comment by casella** [comment]
Online resources can also be an option, these days.

**Comment by dietmarw** [comment]
> Online resources can also be an option, these days.

Though, thinking about it, I would be careful assuming unrestricted access to "western" internet sources. Speaking from the experience of holding modelling workshops in China where I needed to have OpenModelica available on a USB stick then having also the documentation not with it would not be a good thing. 

I think it would be better to have the PDF versions release time of the online resources packaged with OM release. They are already available, not sure why old versions are bundled with the current release.
At the same time an additional link to online resources pointing to https://openmodelica.org/useresresources/userdocumentation could be added.

**Comment by sjoelund** [comment]
Yeah, it could be two entries, or looking to see if the documentation was installed and then not showing the local one. OpenModelica servers are indeed very slow in China (they are faster through VPNs, but those are increasingly restricted).

**Comment by casella** [comment]
> Though, thinking about it, I would be careful assuming unrestricted access to "western" internet sources. Speaking from the experience of holding modelling workshops in China where I needed to have OpenModelica available on a USB stick then having also the documentation not with it would not be a good thing.

Good point. And maybe you want to work from an airplane.

> I think it would be better to have the PDF versions release time of the online resources packaged with OM release. They are already available, not sure why old versions are bundled with the current release. At the same time an additional link to online resources pointing to https://openmodelica.org/useresresources/userdocumentation could be added.

Sounds good

**Comment by casella** [comment]
Adding @anargam to the loop

**Comment by dietmarw** [comment]
I've just revisited the menu again with `1.19.0~dev-737-gc242d85` and updated the [original list of issues](https://github.com/OpenModelica/OpenModelica/issues/8165#issue-1057181694).

**Comment by adeas31** [comment]
Isn't this the [original list of issues](https://github.com/OpenModelica/OpenModelica/issues/8165#issuecomment-976512476)?

See #8785.
- Fixed the typos.
- Changed `OpenModelica User's Guide F1` to point to the online documentation. The offline can be accessed via the pdf `OpenModelica UsersGuide (PDF)`.

The open issues now are,

- [ ] Install `omc-doc` together with OMEdit.
- [ ] Install TLM Simulator.

**Comment by dietmarw** [comment]
It looks like #8785 will replace pointing to the locally installed (and outdated) version to the online version. But as we have discussed before, it would be better to fix the locally installed version with an up to date one. 
So my updated test was after installation of omc-doc and I thought it better to update the top-most ticket comment.

**Comment by adeas31** [comment]
> It looks like https://github.com/OpenModelica/OpenModelica/pull/8785 will replace pointing to the locally installed (and outdated) version to the online version. But as we have discussed before, it would be better to fix the locally installed version with an up to date one.

Yes we need to fix the locally installed. But we were always pointing to the locally installed ones. #8785 gives access to both i.e., online version and locally installed pdf version.

> So my updated test was after installation of omc-doc and I thought it better to update the top-most ticket comment.

I think we should consider the case with installation of omc-doc. The only thing is to get it installed automatically with OMEdit.

**Comment by dietmarw** [comment]
> I think we should consider the case with installation of omc-doc. The only thing is to get it installed automatically with OMEdit.

Yes, and that is what I have done :-)

**Comment by casella** [comment]
@adrpo, can you make sure that omc-doc is a dependency on omc, so it is installed automatically?

**Comment by adrpo** [comment]
I had a talk with @sjoelund and it seems the omc-doc package was not updated since 1.12. We need to change the way we build and ship documentation inside the source tarballs. I will see if I can manage to fix this.

**Comment by adrpo** [comment]
I have read this ticket again and I realized I was talking about the way we handle the Windows documentation.
I have talked with  @sjoelund and we know how to properly fix this by putting the documentation together with the various packages we have and not separately. But building that stage and fixing all the bugs will take time so for 1.19 I will see how I can generate omc-doc-1.19 package on the release chan.

**Comment by casella** [comment]
@adrpo will try to fix this as soon as he has time for that, sorry for the delay.
